Mathlib/Tactic/CancelDenoms/Core.lean

```diff
@@ -328,7 +329,7 @@ open Elab Tactic
 
 def cancelDenominatorsAt (fvar : FVarId) : TacticM Unit := do
   let t ← instantiateMVars (← fvar.getDecl).type
-  let (new, eqPrf) ← CancelDenoms.cancelDenominatorsInType t
+  let (new, eqPrf) ← CancelDenoms.cancelDenominatorsInType t.consumeMData
```

Comment:
I don't think this is the right fix; I think `findCompLemma` should be moved to `MetaM`, and should do a `whnfR`

Reply:
(the current version is also tricked by reducible definitions, not just metadata)

Reply:
Ok, this change means that `≠` does not pass as `Ne`, but as `¬ Eq`: I'll fix this in a bit!

Reply:
A suggested by Eric, I pushed a change with `whnfR` in `findCompLemma` and therefore bypassed the need for `consumeMData/cleanupAnnotations`.